engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -2312,6 +2315,19 @@ double paladin_t::resource_gain( resource_e resource_type, double amount, gain_t
     {
       amount *= 1.0 + buffs.holy_avenger -> data().effectN( 1 ).percent();
     }
+
+    if (
+        legendary.from_dusk_till_dawn -> ok() &&
+        amount > 0 &&
+        resources.current[ RESOURCE_HOLY_POWER ] < resources.base[ RESOURCE_HOLY_POWER ] &&
```

Comment:
It's clearer to use resources.max[] than resources.max[] even if they return the same value.

Reply:
You could also move the whole block below player_t::resource_gain() and only check whether the amount returned by it is != 0 and if you're currently at max.